include/deal.II/lac/diagonal_matrix.h

```diff
@@ -179,6 +179,12 @@ class DiagonalMatrix : public Subscriptor
                    const VectorType &src) const;
 
   /**
+   * Initialize vector @p dst. This is a part of the interface required
+   * by linear operator.
```

Comment:
Also, can you say what "initialization of the vector" actually entails? I suspect you resize it, set it to zero, etc? Or what does that mean?